### ISSUE TITLE
[Fable 2] warning FABLE: Cannot make conversion at compile time because source type is unknown

### ISSUE THREAD

**Issue by MangelMaxime** [fable-compiler/Fable#1585]
### Description

When compiling `Thoth.Elmish.Demo` app I get this warning:

```fs

WARNING in /Users/maximemangel/Workspaces/Github/MangelMaxime/Thot/src/Thoth.Elmish.Toast/Toast.fs
/Users/maximemangel/Workspaces/Github/MangelMaxime/Thot/src/Thoth.Elmish.Toast/Toast.fs(353,39): (353,66) warning FABLE: Cannot make conversion at compile time because source type is unknown
 @ ./src/Toast.fs 20:0-355 124:24-32 124:55-61 143:25-35 149:13-21 158:11-22 163:55-59 163:136-143 163:223-228 163:273-280 170:13-20 179:65-74 182:13-20 189:70-84 197:71-86 207:26-33 207:34-42 207:54-64 212:26-33 212:34-42 212:54-64 217:26-31 217:32-40 217:50-60 307:26-30 307:31-39 307:48-58 312:22-30 312:56-64 312:89-97 312:124-132 312:155-163 312:185-193 355:22-28 355:94-100 355:166-172 355:235-241 429:100-106 432:81-112
 @ ./src/App.fs
 @ ./Thoth.Elmish.Demo.fsproj
 @ multi @babel/polyfill ./Thoth.Elmish.Demo.fsproj

WARNING in entrypoint size limit: The following entrypoint(s) combined asset size exceeds the recommended limit (244 KiB). This can impact web performance.
Entrypoints:
  demo (364 KiB)
      vendors.js
      demo.js
```

The lines mentioned by the warning is:

```fs
let private delayedCmd (notification : Toast<'icon>) =
    match notification.Delay with
    | Some delay ->
        promise {
            do! Promise.sleep (int delay.TotalMilliseconds) // This line
            return notification
        }
    | None -> failwith "No delay attached to notification can't delayed it. `delayedCmd` should not have been called by the program"
```

`delay` is a `TimeSpan` value.

[Source code](https://github.com/MangelMaxime/Thoth/blob/master/src/Thoth.Elmish.Toast/Toast.fs#L353)
